### PR TITLE
Objective-C Test Changes

### PR DESCRIPTION
**Pull request by schmidt-sebastian**
This migrates the Spec Tests and the integration tests to Protobuf

### CODE REVIEW

File: Firestore/Example/Tests/Integration/FSTDatastoreTests.mm [hunk 1]
```diff
@@ -144,8 +144,8 @@ - (void)expectListenEventWithDescription:(NSString *)description {
                                                                     description]]];
 }
 
-- (void)applySuccessfulWriteWithResult:(const MutationBatchResult &)batchResult {
-  _writeEvents.push_back(batchResult);
+- (void)applySuccessfulWriteWithResult:(MutationBatchResult)batchResult {
+  _writeEvents.push_back(std::move(batchResult));
```
Comment (var-const) [inline Firestore/Example/Tests/Integration/FSTDatastoreTests.mm:148]: Well done spotting this!

File: Firestore/Example/Tests/SpecTests/FSTSpecTests.mm [hunk 2]
```diff
@@ -325,17 +323,17 @@ - (SnapshotVersion)parseVersion:(NSNumber *_Nullable)version {
...
 
   XCTAssert([jsonDoc[@"key"] isKindOfClass:[NSString class]]);
-  FieldValue data = [_reader parsedQueryValue:jsonDoc[@"value"]];
-  Document doc = Doc(util::MakeString((NSString *)jsonDoc[@"key"]), version.longLongValue, data,
-                     documentState);
-  return DocumentViewChange{doc, type};
+  google_firestore_v1_Value data = [_reader parsedQueryValue:jsonDoc[@"value"]];
+  MutableDocument doc =
+      Doc(util::MakeString((NSString *)jsonDoc[@"key"]), version.longLongValue, data);
+  if ([options[@"hasLocalMutations"] isEqualToNumber:@YES]) {
```
Comment (var-const) [inline Firestore/Example/Tests/SpecTests/FSTSpecTests.mm:331]: Question: is it necessary to mix numbers and booleans like this? Is there some `isEqualToBoolean` function?

File: Firestore/Example/Tests/SpecTests/FSTSpecTests.mm [hunk 3]
```diff
@@ -506,8 +504,11 @@ - (void)doWriteAck:(NSDictionary *)spec {
                 @"'keepInQueue=true' is not supported on iOS and should only be set in "
                 @"multi-client tests");
 
-  MutationResult mutationResult(version, absl::nullopt);
-  [self.driver receiveWriteAckWithVersion:version mutationResults:{mutationResult}];
+  google_firestore_v1_ArrayValue tansforms{};
```
Comment (var-const) [inline Firestore/Example/Tests/SpecTests/FSTSpecTests.mm:507]: Nit: `s/tansforms/transforms/`.

File: Firestore/Example/Tests/API/FSTUserDataReaderTests.mm [hunk 4]
```diff
@@ -112,38 +124,42 @@ - (void)testConvertsUnsignedCharToInteger {
...
 - (void)testConvertsDates {
   NSArray<NSDate *> *values =
       @[ FSTTestDate(1900, 12, 1, 1, 20, 30), FSTTestDate(2017, 4, 24, 13, 20, 30) ];
   for (NSDate *value in values) {
-    FieldValue wrapped = FSTTestFieldValue(value);
-    XCTAssertEqual(wrapped.type(), FieldValue::Type::Timestamp);
-    XCTAssertEqual(wrapped.timestamp_value(), MakeTimestamp(value));
+    google_firestore_v1_Value wrapped = FSTTestFieldValue(value);
+    XCTAssertEqual(GetTypeOrder(wrapped), TypeOrder::kTimestamp);
+    firebase::Timestamp timestamp = MakeTimestamp(value);
```
Comment (var-const) [inline Firestore/Example/Tests/API/FSTUserDataReaderTests.mm:139]: Please add using declarations for `Timestamp` and `GeoPoint`.

**Review by var-const — CHANGES_REQUESTED** [review]

File: Firestore/Example/Tests/API/FSTUserDataReaderTests.mm (hunk 4 above)
Comment (schmidt-sebastian) [reply Firestore/Example/Tests/API/FSTUserDataReaderTests.mm]: Done

File: Firestore/Example/Tests/SpecTests/FSTSpecTests.mm (hunk 2 above)
Comment (schmidt-sebastian) [reply Firestore/Example/Tests/SpecTests/FSTSpecTests.mm]: Updated to use `boolValue`.

File: Firestore/Example/Tests/SpecTests/FSTSpecTests.mm (hunk 3 above)
Comment (schmidt-sebastian) [reply Firestore/Example/Tests/SpecTests/FSTSpecTests.mm]: fixed

**Review by var-const — APPROVED** [review]

**Pull request merged**